google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/ManagedVmCloudType.java

```diff
@@ -196,6 +204,56 @@ public String getPresentableName() {
       public void setName(String name) {
         this.name = name;
       }
+
+      @NotNull
+      @Override
+      public DeploymentSourceType<?> getType() {
+        return DeploymentSourceType.EP_NAME.findExtension(UserSpecifiedPathDeploymentSourceType.class);
+      }
+    }
+  }
+
+  public static class UserSpecifiedPathDeploymentSourceType extends
+      DeploymentSourceType<ModuleDeploymentSource> {
+    private static final String NAME_ATTRIBUTE = "name";
+
+    public UserSpecifiedPathDeploymentSourceType() {
+      super("filesystem WAR/JAR module");
```

Comment:
It's not something that we need translation since it's sort of a unique internal id or key. For example, IntelliJ's SourceType implementations use names like "module" or "artifact". Do we still want to make it a property?